src/controllers/PostCommentController.js

```diff
@@ -0,0 +1,121 @@
+import { PostComment } from '../models';
+import { statusCodes, responseMessages } from '../constants';
+import { PAGE_LIMIT } from '../constants/shared';
+
+/**
+ * @description PostComment Controller class
+ */
+export default class PostCommentController {
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async createPostComment(req, res) {
+    const { currentUser, body, post } = req;
+    body.author = currentUser._id;
+    body.post = post._id;
+    const postComment = await PostComment.create(body);
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      postComment,
+      message: responseMessages.created('Post Comment'),
+    });
+  }
+
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async updatePostComment(req, res) {
+    const { currentUser, body, postComment } = req;
+
+    if (!currentUser._id.equals(postComment.author)) {
+      return res.status(statusCodes.UNAUTHORIZED).json({
+        status: statusCodes.UNAUTHORIZED,
+        message: responseMessages.unauthorized(),
+      });
+    }
+
+    await postComment.updateOne(body);
+
+    return res.status(statusCodes.OK).json({
+      status: statusCodes.OK,
+      postComment: {
+        ...postComment.toObject(),
+        ...body,
+      },
+      message: responseMessages.updated('Post Comment'),
+    });
+  }
+
+  /**
+   * @author Olivier
+   * @param {Object} req
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async deletePostComment(req, res) {
+    const { currentUser, postComment } = req;
+
+    if (!currentUser._id.equals(postComment.author)) {
+      return res.status(statusCodes.UNAUTHORIZED).json({
+        status: statusCodes.UNAUTHORIZED,
+        message: responseMessages.unauthorized(),
+      });
+    }
+
+    await postComment.deleteOne();
```

Comment:
@oesukam, Won't it be a good idea just not change the status of a comment, to avoid deleting the actual data?

Reply:
@gracelungu Yes, we can just update the status. I am updating it